src/entities/Mob.ts

```diff
@@ -67,9 +75,23 @@ export class Mob {
     }
 
     public fireProjectile() {
-        if(gameManager.isGameRunning)
+        if(gameManager.isGameRunning && this.isFiring)
             new Projectile(new Vector2(this.mesh.position.x, this.mesh.position.y), character.current)
     }
 
+    public receiveProjectile(projectileDamage: number) {
+        this.health = this.health - projectileDamage;
+        console.log(this.health);
+        if(this.health <= 0) {
+        this.killMob();
+        }
+    }
 
+    // Il faudra faire un truc plus propre
+    killMob() {
+        mobs.slice(this.id);
```

Comment:
Use splice instead of slice. See https://stackoverflow.com/questions/37601282/javascript-array-splice-vs-slice

Reply:
frerot c'est outdated